cypress/integration/collections_spec.js

```diff
@@ -0,0 +1,116 @@
+import { shouldBeRedirectedToLogin } from '../support/assertions';
+
+describe('Dashboard Collections Page', () => {
+  describe('When not logged in', () => {
+    it('should redirect to login page', () => {
+      cy.visit('/#/collections');
+      shouldBeRedirectedToLogin();
+    });
+  });
+
+  describe('When logged in', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.task('resetState');
+    });
+
+    after(() => {
+      cy.task('resetState');
+    });
+
+    it('displays a link to view collections', () => {
+      cy.visit('/');
+
+      cy.contains('nav li a', 'Collections').should('exist').as('collections');
+      cy.get('@collections').should('have.attr', 'href', '#/collections');
+      cy.get('@collections').click();
+
+      cy.url().should('include', 'collections');
+      cy.contains('.heading--xlarge', 'Collections');
+
+      cy.get('table tbody tr').its('length').should('be.eq', 5);
+    });
+
+    it('collections page displays a button to add a new collection', () => {
+      const name = 'TESTCOLLECTION';
+      const version = '006';
+      cy.visit('/#/collections');
+
+      cy.contains('.heading--large', 'Collection Overview');
+      cy.contains('a', 'Add a Collection').should('exist').as('addCollection');
+      cy.get('@addCollection').should('have.attr', 'href', '#/collections/add');
+      cy.get('@addCollection').click();
+
+      // fill the form and submit
+      const collection = '{{}"name":"TESTCOLLECTION","version":"006","dataType":"TESTCOLLECTION", "duplicateHandling":"error"}';
+      cy.get('textarea').type(collection, { force: true });
+      cy.get('form').get('input').contains('Submit').click();
+
+      // displays the new collection
+      cy.contains('.heading--xlarge', 'Collections');
+      cy.contains('.heading--large', `${name} / ${version}`);
+      cy.url().should('include', `#/collections/collection/${name}/${version}`);
+
+      // verify the collection's properties by looking at the Edit page
+      cy.contains('a', 'Edit').should('exist').click();
+      cy.get('form .ace_content')
```

Comment:
How are you sure that this is testing the state after the collection was edited and not before? I know that there is a "click" command above but I don't think it guarantees the page is fully reloaded before the next command. 

It seems like it might be preferable to test the contents of the "view" page and not the "edit" page

Reply:
@markdboyd Which "view" page are you preferring to?  The overview page of the collection doesn't the information we are looking for.  There is a `Definition` page which is similar to the `Edit` page.

Reply:
Nevermind, I was misunderstanding